sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/HttpClientProviders.java

```diff
@@ -34,4 +36,19 @@ public static HttpClient createInstance() {
         }
         return defaultProvider.createInstance();
     }
+
+    /**
+     * Returns a list of all {@link HttpClient HttpClients} that are discovered in the classpath.
+     *
+     * @return A list of all {@link HttpClient HttpClients} discovered in the classpath.
+     */
+    public static List<HttpClient> getAllHttpClients() {
```

Comment:
I don't think we would want this to be public API

Reply:
This is decided by Srikanta and Jonathan. @srnagar introduce the logic here. We want to expose the clients to test framework, so we are able to get all clients from test without manual adding clients in test every time. It is under implementation, the only pain is to expose the API to azure-core-test. I think we agree to approach in this way.